### PR TITLE
Sync Gateway CBG-365 - Update logging config for SG 2.1.0+

### PR DESCRIPTION
**Pull request by bbrks**
## Changes
- Added a product version filter, that can be used to skip generation of files for particular versions (matched by per-product regular expression).
- Updated the default config and `entrypoint.sh` script for SG 2.1.0+ for the added `defaultLogFilePath` parameter.
- Did a big update on README.md for SG 2.1.0 and up for more up to date information and examples
  - [x] @jamiltz could you take a read through the changes in `generate/resources/sync-gateway/README.md` when you get a chance to review?

### CODE REVIEW

**Comment by bbrks** [comment]
/cc @JFlath :
This update adds the `--defaultLogFilePath` flag to the container, to enable support file logging by default.
The default built-in config also enables console logging, to provide typical 'docker log' behaviour in the default/quickstart path, unless overridden by a custom config.

File: community/sync-gateway/2.1.0/README.md [hunk 1]
```diff
@@ -1,83 +1,84 @@
...
+
+## Running Sync Gateway with Docker
 
 ```
-$ docker run -p 4984:4984 -d couchbase/sync-gateway
+$ docker run -d --name sgw -p 4984:4984 couchbase/sync-gateway
 ```
 
-At this point you should be able to run a curl request against the running Sync Gateway on the port 4984 public port:
+At this point you should be able to send a HTTP request against Sync Gateway public port `4984` using curl:
```
Comment (JFlath) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion
At this point you should be able to send a HTTP request to the Sync Gateway public port `4984` using curl:
```

File: community/sync-gateway/2.1.0/README.md [hunk 2]
```diff
@@ -86,117 +87,82 @@ $ mv basic-walrus-bucket.json my-sg-config.json
...
+**Step - 2 :** Run Sync Gateway and use that configuration file:
 
-Sync Gateway can also load it's configuration directly from a URL.
+`$ docker run -p 4984:4984 -d -v /tmp:/tmp/config couchbase/sync-gateway /tmp/config/my-sg-config.json`
 
-First upload a configuration file to a publicly available hosting site of your choice (Amazon S3, Github, etc)
+## Using a URL
 
-Then start Sync Gateway and give it the URL to the raw JSON data:
+Sync Gateway can also load it's configuration directly from a public URL.
```
Comment (JFlath) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion
Sync Gateway can also load its configuration directly from a public URL.
```

File: community/sync-gateway/2.1.0/README.md [hunk 3]
```diff
@@ -86,117 +87,82 @@ $ mv basic-walrus-bucket.json my-sg-config.json
...
-Run Couchbase Server in a docker container, and put it in the `couchbase` network.
-
-```
-$ docker run --net=couchbase -d --name couchbase-server -p 8091-8094:8091-8094 -p 11210:11210 couchbase
-```
-
-Now go to the Couchbase Server Admin UI on [http://localhost:8091](http://localhost:8091) (on OSX, replace localhost with the docker machine host IP) and go through the Setup Wizard.  See [Couchbase Server on Dockerhub](https://hub.docker.com/r/couchbase/server/) for more info.
-
-Create a `/tmp/my-sg-config.json` file on your host machine, with the following:
+**Step - 3 :** Create a configuration file as described in the above config section, and customise the server property::
```
Comment (JFlath) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion
**Step - 3 :** Create a configuration file as described in the above config section, and customise the server property:
```

File: community/sync-gateway/2.1.0/README.md [hunk 4]
```diff
@@ -86,117 +87,82 @@ $ mv basic-walrus-bucket.json my-sg-config.json
...
 
-```
-$ docker logs container-id > /tmp/sync_gateway.log 2>&1
-$ docker exec container-id mkdir -p /var/log/sync_gateway/
-$ docker cp /tmp/sync_gateway.log contaner-id:/var/log/sync_gateway/sync_gateway_error.log
-```
+`# curl -X POST http://localhost:4985/_sgcollect_info -H 'Content-Type: application/json' -d '{"upload": false, "customer": "your-company-name"}'`
 
-Once that is done, you can run `sgcollect_info` via:
+Set upload to `true` along with a customer name if you wish to automatically upload the collection to Couchbase Support.
```
Comment (JFlath) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion

```

Tempted to cut this down a bit, and leave it to point to the main docs. Lowers the risk of random uploads/doesn't set the expectation that Support will engage any user.

File: community/sync-gateway/2.1.0/README.md [hunk 5]
```diff
@@ -1,83 +1,84 @@
 
-Sync Gateway is REST API server that allows Couchbase Lite mobile databases to synchronize data. It can also be used as a standalone data storage system.
+This README will guide you through running Couchbase Sync Gateway with Docker Containers.
 
-For more information, see the [Couchbase Mobile Overview](http://developer.couchbase.com/mobile).
+[Sync Gateway](https://www.couchbase.com/products/sync-gateway) is a horizontally scalable web server that securely manages the access control and synchronization of data between [Couchbase Lite](https://www.couchbase.com/products/lite) and [Couchbase Server](https://www.couchbase.com/products/server).
 
-## Quickstart
+For configuration references, or additional information about anything described below, visit the [Sync Gateway documentation site](https://docs.couchbase.com/sync-gateway/2.5/index.html).
```
Comment (jamiltz) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion
For configuration references, or additional information about anything described below, visit the [Sync Gateway documentation site](https://docs.couchbase.com/sync-gateway/current/index.html).
```

File: community/sync-gateway/2.1.1/README.md [hunk 6]
```diff
@@ -86,117 +87,82 @@ $ mv basic-walrus-bucket.json my-sg-config.json
...
+You can find more information about the parameters used in this request in the [sgcollect_info documentation](https://docs.couchbase.com/sync-gateway/current/admin-rest-api.html#/server/post__sgcollect_info).
 
-## Support
 
-[Couchbase Forums](https://forums.couchbase.com/)
-
-## Licensing
+# Licensing
 
 Sync Gateway comes in 2 Editions: Enterprise Edition and Community Edition. You can find details on the differences between the 2 and licensing details on the [Sync Gateway Editions](http://developer.couchbase.com/documentation/server/4.5/introduction/editions.html) page.
```
Comment (jamiltz) [inline community/sync-gateway/2.1.1/README.md]:
```suggestion
Sync Gateway comes in 2 Editions: Enterprise Edition and Community Edition. You can find details on the differences between the 2 and licensing details on the [Sync Gateway Editions](https://docs.couchbase.com/server/current/introduction/editions.html) page.
```

File: community/sync-gateway/2.1.0/README.md [hunk 7]
```diff
@@ -86,117 +87,80 @@ $ mv basic-walrus-bucket.json my-sg-config.json
...
+      "username": "Administrator",
+      "password": "password",
       "users": { "GUEST": { "disabled": false, "admin_channels": ["*"] } }
     }
   }
 }
 ```
 
-Start a Sync Gateway container in the `couchbase` network and use the `/tmp/my-sg-config.json` file:
+**Step - 3 :** Start a Sync Gateway container in the `couchbase` network and use the configuration file you just wrote:
```
Comment (jamiltz) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion
**Step - 4 :** Start a Sync Gateway container in the `couchbase` network and use the configuration file you just wrote:
```

File: community/sync-gateway/2.1.0/README.md [hunk 8]
```diff
@@ -1,83 +1,84 @@
...
+## Exposing admin port to host
 
-If you need to expose port 4985 to the host machine, you can do so with the following steps.
+Although not recommended for security reasons, if you need to expose the admin port to the host machine, you can do so with the following steps.
 
-You may want to stop any currently running Sync Gateway containers with `docker stop container-id`.
+**Step - 1 :** Stop any currently running Sync Gateway containers:
 
-Start a container with these arguments:
+`docker stop sgw`.
```
Comment (jamiltz) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion
`docker stop sgw`
```

File: community/sync-gateway/2.1.0/README.md [hunk 9]
```diff
@@ -86,117 +87,80 @@ $ mv basic-walrus-bucket.json my-sg-config.json
...
 
-[Couchbase Forums](https://forums.couchbase.com/)
 
-## Licensing
+# Licensing
 
 Sync Gateway comes in 2 Editions: Enterprise Edition and Community Edition. You can find details on the differences between the 2 and licensing details on the [Sync Gateway Editions](http://developer.couchbase.com/documentation/server/4.5/introduction/editions.html) page.
 
--	Enterprise Edition -- free for development, testing and POCs. Requires a paid subscription for production deployment. Please refer to the [subscribe](http://www.couchbase.com/subscriptions-and-support) page for details on enterprise edition agreements.
+- **Enterprise Edition** -- free for development, testing and POCs. Requires a paid subscription for production deployment. Please refer to the [subscribe](https://www.couchbase.com/subscriptions-and-support) page for details on enterprise edition agreements.
```
Comment (jamiltz) [inline community/sync-gateway/2.1.0/README.md]:
```suggestion
- **Enterprise Edition** -- free for development, testing and POCs. Requires a paid subscription for production deployment. Please refer to the [subscriptions](https://www.couchbase.com/subscriptions-and-support) page for details on enterprise edition agreements.
```

**Comment by bbrks** [comment]
@kimtrang I think the PR is ready for your review. Thanks!

**Comment by ceejatec** [comment]
I'd really rather not submit this as-is. Depending on someone to manually remember which changed files to commit will fail almost immediately. We need to somehow update the generator to create the right files, or at least suppress it from creating wrong ones.

**Comment by bbrks** [comment]
@ceejatec That's fair - I'll take a look at how to achieve this reasonably nicely in the generator script.

**Comment by ceejatec** [comment]
There's code in there already (ugly code) to change certain variables based on the product version, and then those variables are substituted in the templates. That's useful for things like "use a different Docker base image" or "add extra apt-get packages", but for a whole different README and such it probably isn't.

I'm guessing that we probably *generally* want to leave the older generated files alone, so perhaps changing the generator to just skip generating files for anything older than 2.1.0 would work? It's not super-maintainable, but it'd get the job done for now.

**Comment by bbrks** [comment]
@ceejatec I've added a regular expression filter per-product that can be used to skip generation of particular versions.

```go
skipGeneration = ProductVersionFilter{
	ProductSyncGw: regexp.MustCompile(`^(1\.|2\.0\.).+$`), // 1.x and 2.0.x
}
```

When run, it logs which versions were skipped:

```
...
2019/05/15 12:02:28 Skipping generation for sync-gateway enterprise 1.5.1
2019/05/15 12:02:28 Skipping generation for sync-gateway enterprise 2.0.0
2019/05/15 12:02:28 Skipping generation for sync-gateway enterprise 2.0.0-devbuild
2019/05/15 12:02:28 generateDockerfile called with: {enterprise sync-gateway 2.1.0 false}
2019/05/15 12:02:28 generateDockerfile called with: {enterprise sync-gateway 2.1.1 false}
...
```

**Comment by ceejatec** [comment]
Ok, I'm good with that solution at least for now. @JFlath are you on board now?

**Comment by JFlath** [comment]
Sounds good to me :)

**Comment by bbrks** [comment]
@ceejatec are we OK to merge this?

**Comment by ceejatec** [comment]
Sure thing - done. Thanks!

**Pull request merged**